generator/js_generator.cc

```diff
@@ -263,14 +263,14 @@ char ToLowerASCII(char c) {
 std::vector<std::string> ParseLowerUnderscore(const std::string& input) {
   std::vector<std::string> words;
   std::string running = "";
-  for (int i = 0; i < input.size(); i++) {
-    if (input[i] == '_') {
+  for (auto c : input) {
```

Comment:
style tweak: use const auto& or auto& instead to explicitly signal as appropriate.

Reply:
Maybe it doesn't matter, but that does not seem to be the existing style of this file. E.g.:
https://github.com/protocolbuffers/protobuf-javascript/blob/3ff6090f139d71453062fb96c66e9aff801709c2/generator/js_generator.cc#L391 https://github.com/protocolbuffers/protobuf-javascript/blob/3ff6090f139d71453062fb96c66e9aff801709c2/generator/js_generator.cc#L397 https://github.com/protocolbuffers/protobuf-javascript/blob/3ff6090f139d71453062fb96c66e9aff801709c2/generator/js_generator.cc#L1762 https://github.com/protocolbuffers/protobuf-javascript/blob/3ff6090f139d71453062fb96c66e9aff801709c2/generator/js_generator.cc#L3355

In general, this we find ourselves iterating over integers or pointers, which I think should be fine to iterate by value?

Reply:
Generally speaking, our style is to add & for object types. I think I misread the thing I tagged this on. I meant to do it on something that was a std::string. Similar, we should annotate pointers with auto*|const auto*.

Technically speaking, I think auto should do the right thing most of the time. It's just useful context to the reader and maybe an occasional nudge to auto.

I didn't appreciate that the surrounding code may not follow this guidance, so I'm ok taking the changes as is.